src/meta-srv/src/procedure/region_failover/deactivate_region.rs

```diff
@@ -42,6 +43,35 @@ impl DeactivateRegion {
         Self { candidate }
     }
 
+    async fn mark_leader_downgraded(
+        &self,
+        ctx: &RegionFailoverContext,
+        failed_region: &RegionIdent,
+    ) -> Result<()> {
+        let table_id = failed_region.table_id;
+
+        let table_route_value = ctx
+            .table_metadata_manager
+            .table_route_manager()
+            .get(table_id)
+            .await
+            .context(error::TableMetadataManagerSnafu)?
+            .context(error::TableRouteNotFoundSnafu { table_id })?;
+
+        ctx.table_metadata_manager
+            .update_leader_region_status(table_id, table_route_value, |region| {
+                if region.region.id.region_number() == failed_region.region_number {
```

Comment:
If the region is already marked as "downgraded", it doesn't need to be marked again. Otherwise the table route metadata will be updated unnecessarily.